wpilibc/src/main/native/cpp/simulation/Mechanism2D.cpp

```diff
@@ -0,0 +1,39 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2020 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#include "frc/simulation/Mechanism2D.h"
+
+#include <wpi/raw_ostream.h>
+
+using namespace frc;
+
+Mechanism2D::Mechanism2D() : m_device{"Mechanism2D"} {}
+
+void Mechanism2D::SetLigamentAngle(std::string ligamentPath, float angle) {
+  if (m_device) {
+    ligamentPath = ligamentPath + "angle/";
```

Comment:
Since you're appending... you should change the ligamentPath argument to be `const wpi::Twine& ligamentPath`, then you can do the following:
```
wpi::SmallString<64> fullPathBuf;
wpi::StringRef fullPath = (ligamentPath + "angle/").toStringRef(fullPathBuf);
```
and use fullPath instead of ligamentPath in the rest of the function.
This will keep the string concatenation on the stack rather than allocating memory.